File: api/nuelink.js
const express = require('express');
const { exec } = require('child_process');
const cors = require('cors');
const app = express();

app.use(cors());

// Ye endpoint ab aapke file name se match karega
app.get('/api/nuelink', (req, res) => {
    const videoUrl = req.query.url;

    console.log(`📥 Request Received for: ${videoUrl}`);

    if (!videoUrl) {
        return res.status(400).json({ success: false, message: "URL missing!" });
    }

    // Pro command to get high quality video
    const command = `python3 -m yt_dlp --dump-json --no-check-certificate --format "bestvideo[ext=mp4]+bestaudio[ext=m4a]/best[ext=mp4]/best" "${videoUrl}"`;

    exec(command, (error, stdout, stderr) => {
        if (error) {
            console.error(`❌ Scraper Error: ${stderr}`);
            return res.status(500).json({ success: false, message: "Scraping failed." });
        }

        try {
            const metadata = JSON.parse(stdout);
            res.json({
                success: true,
                data: {
                    title: metadata.title,
                    thumbnail: metadata.thumbnail,
                    url: metadata.url, // Main Video URL
                    uploader: metadata.uploader,
                    duration: metadata.duration_string
                }
            });
            console.log(`✅ Done: ${metadata.title}`);
        } catch (e) {
            res.status(500).json({ success: false, message: "JSON parsing error." });
        }
    });
});

const PORT = process.env.PORT || 8000;
app.listen(PORT, () => console.log(`🚀 API is running on port ${PORT}`));

File: api/nuelink.py
from http.server import BaseHTTPRequestHandler
import json
import yt_dlp
from urllib.parse import urlparse, parse_qs

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        # 1. URL se query parameters nikalna
        query = urlparse(self.path).query
        params = parse_qs(query)
        video_url = params.get('url', [None])[0]

        if not video_url:
            self.send_response(400)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({"success": False, "message": "URL missing!"}).encode())
            return

        # 2. yt-dlp ki settings
        ydl_opts = {
            'format': 'bestvideo[ext=mp4]+bestaudio[ext=m4a]/best[ext=mp4]/best',
            'quiet': True,
            'no_warnings': True,
        }

        try:
            with yt_dlp.YoutubeDL(ydl_opts) as ydl:
                # 3. Scraping logic
                info = ydl.extract_info(video_url, download=False)
                
                response_data = {
                    "success": True,
                    "data": {
                        "title": info.get('title'),
                        "thumbnail": info.get('thumbnail'),
                        "url": info.get('url'),
                        "uploader": info.get('uploader'),
                        "duration": info.get('duration_string')
                    }
                }

                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.send_header('Access-Control-Allow-Origin', '*') # CORS allow
                self.end_headers()
                self.wfile.write(json.dumps(response_data).encode())

        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({"success": False, "error": str(e)}).encode())
